src/devices/cpu/t11/t11.cpp

```diff
@@ -210,6 +212,123 @@ static const struct irq_table_entry irq_table[] =
 	{ 7<<5, 0140 }
 };
 
+// PSW7 masks external interrupts (except IRQ1) -- IRQ2, IRQ3, VIRQ
+// PSW11 also masks IRQ1.  PSW10 also masks ACLO.
+void k1801vm1_device::t11_check_irqs()
+{
+	// 2. bus error on vector fetch; nm, vec 160012
+	// 3. double bus error; nm, vec 160006
+	// 4. bus error; PSW11, PSW10
+	if (m_bus_error)
+	{
+		m_bus_error = false;
+		m_mcir = MCIR_IRQ;
+		m_vsel = T11_TIMEOUT;
+	}
+	// 5. illegal insn; nm
+	else if (m_mcir == MCIR_ILL)
+	{
+		WWORD(VM1_SEL1, RWORD(VM1_SEL1) & ~SEL1_HALT);
+		PUSH(PSW);
+		PUSH(PC);
+		PC = RWORD(m_vsel);
+		PSW = RWORD(m_vsel + 2);
+	}
+	// 6. trace trap; WCPU
+	else if (m_trace_trap && m_mcir == MCIR_NONE) // allow trap_to() to execute first
+	{
+		if (GET_T)
+		{
+			m_mcir = MCIR_IRQ;
+			m_vsel = T11_BPT;
+		}
+		else
+			m_trace_trap = false;
+	}
+	else if (GET_T)
+	{
+		m_trace_trap = true;
+	}
+	// 7. power fail; PSW10
+	else if (m_power_fail)
+	{
+		m_mcir = MCIR_IRQ;
+		m_vsel = T11_PWRFAIL;
+	}
+	// 8. external HALT (nIRQ1 pin); PSW11, PSW10
+	else if (m_hlt_active)
+	{
+		m_mcir = MCIR_HALT;
+		m_vsel = VM1_HALT;
+	}
+	// 9. internal timer, vector 0270; PSW7, PSW10
+	// 10. line clock (nIRQ2 pin); PSW7, PSW10
+	else if (BIT(m_cp_state, 2) && !GET_I)
+	{
+		m_mcir = MCIR_IRQ;
+		m_vsel = VM1_EVNT;
+	}
+	// 11. nIRQ3 pin; PSW7, PSW10
+	else if (BIT(m_cp_state, 3) && !GET_I)
+	{
+		m_mcir = MCIR_IRQ;
+		m_vsel = VM1_IRQ3;
+	}
+
+	// 12. nVIRQ pin; PSW7, PSW10
+	else if (m_vec_active && !GET_I)
+	{
+		int vec = m_in_iack_func(0);
+		if (vec == -1 || vec == 0)
```

Comment:
Please don’t leave blank lines between the end of an `if` block and a following `else` – it greatly increases the risk of confusion when reading the code.